modules/fluid_properties/src/fluidproperties/IdealRealGasMixtureFluidProperties.C

```diff
@@ -14,8 +14,224 @@
+                                                           std::vector<Real> & dy_dx) const        \
+  {                                                                                                \
+    const Real x_primary = primaryMassFraction(x);                                                 \
+                                                                                                   \
+    Real y_primary, dy_dT_primary, dy_dv_primary;                                                  \
+    _fp_primary->prop##_from_T_v(T, v / x_primary, y_primary, dy_dT_primary, dy_dv_primary);       \
+    y = x_primary * y_primary;                                                                     \
+    dy_dT = x_primary * dy_dT_primary;                                                             \
+    dy_dv = dy_dv_primary;                                                                         \
+                                                                                                   \
+    Real dy_dT_sec;                                                                                \
+    std::vector<Real> y_sec(_n_secondary_vapors), dy_dv_sec(_n_secondary_vapors);                  \
+    for (unsigned int i = 0; i < _n_secondary_vapors; i++)                                         \
+    {                                                                                              \
+      _fp_secondary[i]->prop##_from_T_v(T, v / x[i], y_sec[i], dy_dT_sec, dy_dv_sec[i]);           \
+      y += x[i] * y_sec[i];                                                                        \
+      dy_dT += x[i] * dy_dT_sec;                                                                   \
+      dy_dv += dy_dv_sec[i];                                                                       \
+    }                                                                                              \
+                                                                                                   \
+    dy_dx.resize(_n_secondary_vapors);                                                             \
+    for (unsigned int i = 0; i < _n_secondary_vapors; i++)                                         \
+    {                                                                                              \
+      dy_dx[i] = y_sec[i] - x[i] * dy_dv_sec[i] * v / (x[i] * x[i]);                               \
+      for (unsigned int j = 0; j < _n_secondary_vapors; j++)                                       \
+      {                                                                                            \
+        if (j == i)                                                                                \
+          continue;                                                                                \
+        const Real dxj_dxi = -x[j] / (1. - x[i]);                                                  \
```

Comment:
so xj = 1 -xi - other things (roughly)
so dxj / dxi = -1 ?

how did you get that

Reply:
I'm very confused/frustrated at this line. I vaguely remember figuring out this thing and even putting a comment, but then I couldn't figure out how to put a comment in the macro. But now the only answer I can come up with is that this should be 0 for `j != i`.

Reply:
This is with `x_p + \sum_j^N x_j = 1`, where `N` is the number of secondary components. So the secondary mass fractions are independent; the primary mass fraction would take the change. I guess I need to test it.